### PR TITLE
Patch django legacy version 1.4

### PR DESCRIPTION
**Pull request by labbati**
Django 1.4 had a initialization flow quite different if compared to the minimum django version we currently provides, i.e. 1.8. Specifically django misses a bootstrap hook as the `AppConfig.ready` method that we use in our current django patching mechanism.

This PR adds a legacy patch mechanism not based on the `AppConfig.ready` method, whit the companion test suite.

### CODE REVIEW

File: tests/contrib/django_legacy/utils.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    def tearDown(self):
+        # empty the tracer spans from test operations
+        self.tracer.writer.spans = []
+        self.tracer.writer.pop_traces()
+        self.unpatch()
+
+    def patch(self):
+        patch()
+
+    def activate_db_patch_for_non_request_based_tests(self):
```
Comment (Kyle-Verhoog) [inline tests/contrib/django_legacy/utils.py]: :+1:

File: tests/contrib/django_legacy/utils.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+from functools import wraps
+from ddtrace.tracer import Tracer
+from ddtrace.contrib.django.conf import settings
+from ddtrace.contrib.django.legacy import patch
+from ddtrace.contrib.django.db import patch_db, unpatch_db
+from ddtrace.contrib.django.cache import unpatch_cache
+from ddtrace.contrib.django.legacy.patch import DD_DJANGO_PATCHED_FLAG
+from ddtrace.contrib.django.templates import unpatch_template
+from ddtrace.contrib.django.middleware import remove_exception_middleware, remove_trace_middleware
+from ddtrace.contrib.django.legacy import patch as _ready
```
Comment (Kyle-Verhoog) [inline tests/contrib/django_legacy/utils.py]: This is left over from me, we can just use `patch()` as it's imported above.

File: tests/contrib/django_legacy/test_tracing_disabled.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# 3rd party
+from django.test import TestCase
+
+# project
+from ddtrace.tracer import Tracer
+from ddtrace.contrib.django.conf import settings
+from ddtrace.contrib.django.legacy import patch as _ready
```
Comment (Kyle-Verhoog) [inline tests/contrib/django_legacy/test_tracing_disabled.py]: Left over from me: we can leave the import as `patch` no need to call it `_ready`

**Comment by labbati** [comment]
Just rebase on `0.16-dev`, not be merged for now

**Pull request closed without merging**